lib/kubernetes-deploy/runner.rb

```diff
@@ -282,10 +283,12 @@ def validate_namespace
     def run_kubectl(*args, namespaced: true, with_context: true)
       args = args.unshift("kubectl")
       if namespaced
+        raise FatalDeploymentError, "Namespace missing for namespaced command" if @namespace.blank?
         args.push("--namespace=#{@namespace}")
       end
 
       if with_context
+        raise FatalDeploymentError, "Explicit context is required to run this command" if @context.blank?
```

Comment:
> These are errors that people will see in deploy logs right? 

Yes, they could, but if they do they need to come to us to fix it. The real user-facing validations are here: https://github.com/Shopify/kubernetes-deploy/blob/master/lib/kubernetes-deploy/runner.rb#L218-L224. This is an extra safety layer because of the danger of deploying to the wrong place, especially during development.